cnf-certification-test/performance/suite.go

```diff
@@ -232,30 +233,86 @@ func testSchedulingPolicyInCPUPool(env *provider.TestEnvironment,
 	testhelper.AddTestResultReason(compliantContainersPids, nonCompliantContainersPids, tnf.ClaimFilePrintf, ginkgo.Fail)
 }
 
+func getExecProbesCmds(c *provider.Container) map[string]bool {
+	cmds := map[string]bool{}
+
+	if c.LivenessProbe != nil && c.LivenessProbe.Exec != nil {
+		cmd := strings.Join(c.LivenessProbe.Exec.Command, "")
+		cmd = strings.Join(strings.Fields(cmd), "")
+		cmds[cmd] = true
+	}
+
+	if c.ReadinessProbe != nil && c.ReadinessProbe.Exec != nil {
+		cmd := strings.Join(c.ReadinessProbe.Exec.Command, "")
+		cmd = strings.Join(strings.Fields(cmd), "")
+		cmds[cmd] = true
+	}
+
+	if c.StartupProbe != nil && c.StartupProbe.Exec.Command != nil {
+		cmd := strings.Join(c.StartupProbe.Exec.Command, "")
+		cmd = strings.Join(strings.Fields(cmd), "")
+		cmds[cmd] = true
+	}
+
+	return cmds
+}
+
+const noProcessFoundErrMsg = "No such process"
+
 func testRtAppsNoExecProbes(env *provider.TestEnvironment, cuts []*provider.Container) {
 	var compliantObjects []*testhelper.ReportObject
 	var nonCompliantObjects []*testhelper.ReportObject
 	for _, cut := range cuts {
+		if !cut.HasExecProbes() {
+			compliantObjects = append(compliantObjects, testhelper.NewContainerReportObject(cut.Namespace, cut.Podname, cut.Name, "Container does not define exec probes", true))
+			continue
+		}
+
 		processes, err := crclient.GetContainerProcesses(cut, env)
```

Comment:
This function returns the full list of processes, including the ones that are spawed by a parent process. For instance in the output below:
`/ # ps -e -o pid,ppid,args
PID   PPID  COMMAND
    1     0 /bin/sh -c touch /tmp/healthy;  sleep 200000
    7     1 sleep 200000
 2779     0 sh
 3001     0 sh -c echo test && sleep 1
 3007  3001 sleep 1
 3008  2779 ps -e -o pid,ppid,args
`
The liveliness command ` 3001     0 sh -c echo test && sleep 1`  also spawns ` 3007  3001 sleep 1`. We will check both commands but the second one (`sleep 1`) will not match, the liveliness command in spec. 
I would skip any commands with a parent PID (ppid) matching a exec probe PID. We should add the ppid in the Process struture

Reply:
I see, thanks for pointing that out. I think that the consideration of this particular scenario (which is already covered more generally in the original fix) is getting too complex to be addressed in this PR, so I prefer to merge this one to have this test fixed as soon as possible and implement that improvement in another PR.